mmda/predictors/hf_predictors/token_classification_predictor.py

```diff
@@ -0,0 +1,137 @@
+from typing import List, Union, Dict, Any, Tuple, Optional
+from abc import abstractmethod
+
+from tqdm import tqdm
+from vila.predictors import (
+    SimplePDFPredictor,
+    LayoutIndicatorPDFPredictor,
+    HierarchicalPDFPredictor,
+)
+
+from mmda.types.names import *
+from mmda.types.annotation import Annotation, Span, SpanGroup
+from mmda.types.document import Document
+from mmda.predictors.hf_predictors.utils import (
+    convert_document_page_to_pdf_dict,
+    convert_sequence_tagging_to_spans,
+)
+from mmda.predictors.hf_predictors.base_hf_predictor import BaseHFPredictor
+
+
+class BaseSinglePageTokenClassificationPredictor(BaseHFPredictor):
+
+    REQUIRED_BACKENDS = ["transformers", "torch", "vila"]
+    REQUIRED_DOCUMENT_FIELDS = [Pages, Tokens]
+    DEFAULT_SUBPAGE_PER_RUN = 2
+
+    @property
+    @abstractmethod
+    def VILA_MODEL_CLASS(self):
```

Comment:
why do we need this?

Reply:
this is used in `SinglePageTokenClassificationPredictor`, `IVILATokenClassificationPredictor`, `HVILATokenClassificationPredictor`  to initialize the models -- see line 51 in this file.